### PR TITLE
Add autoLink to phone number 116 117 in Request TAN (EXPOSUREAPP-5007) (closes #2291)

### LINKED ISSUE #2291: Medical Emergency Service Hotline 116 117 not linked as phone in Request TAN
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Samsung Galaxy A5 (2017) SM-A520F
- Android version: 8.0.0
- App version: 1.11.0 (also checked with branch release/1.12.x)

## Describe the bug

In the Request TAN view the telephone number 116 117 is not linked as a phone number.
For comparison, the same phone number is linked in the view shown via App Information > Technical hotline.

## Steps to reproduce the issue

Note: autoLink used in CWA version 1.11.0 is dependent on the device's locale setting. The following steps are with the device's language set to English (United Kingdom). If the device's language is set to English (United States) the telephone number 116 117 is not linked in the Technical Hotline view nor in the Request TAN view.

1. Under "Have you been tested?" tap "NEXT STEPS"
2. Scroll down to "No TAN yet?" and tap
3. Scroll down to the paragraph "If you have any health-related questions, please contact your general practitioner or the medical emergency service hotline, telephone: 116 117."
4. Tap on the phone number 116 117 and note that nothing happens.

Request TAN

---
Compare to the link to phone number 116 117 shown on the Technical hotline view

Technical hotline

## Expected behaviour

After opening the view Request TAN through "Have you been tested?" > "NEXT STEPS" > "No TAN yet?" the phone number 116 117 should be underlined and tapping on it should bring up the Phone app.

The Request TAN view should show the phone number 116 117 the same way that the Technical Hotline view (accessed through three-dot overview menu > App Information > Technical hotline) shows it.

## Possible Fix

Add the following attributes to 
TextView +id/submission_contact_body_other
in layout\include_submission_contact.xml

so that the TextView is aligned with 
TextView @+id/information_contact_body_other
in layout\fragment_information_contact.xml
```
android:autoLink="phone"
android:textColorLink="@color/colorTextTint"

```

~~I will submit a PR for this.~~
PR https://github.com/corona-warn-app/cwa-app-android/pull/2292

## Additional context

This came to light due to examining phone number autoLinking under https://github.com/corona-warn-app/cwa-app-android/issues/2275

---
Internal tracking ID: [EXPOSUREAPP-5007](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5007)
This issue will be solved with EXPOSUREAPP-4992.

### PR DESCRIPTION
**Pull request by MikeMcC399**
### Description

This PR links the Medical Emergency Service Hotline 116 117 in the Request TAN view so that tapping it will bring up the Phone app. This aligns the view with the presentation of the same phone number in the Technical Hotline view.

Issue #2291 describes the problem with screen shots.

### Steps to reproduce

After applying the fix:

1. Under "Have you been tested?" tap "NEXT STEPS"
2. Scroll down to "No TAN yet?" and tap
3. Scroll down to the paragraph "If you have any health-related questions, please contact your general practitioner or the medical emergency service hotline, telephone: 116 117."
4. Note that the phone number 116 117 is shown in the accent color and is underlined
5. Tap on the phone number 116 117 and the phone number is now showing in the Phone app ready to dial 116117.

Request TAN with autoLink

---
Internal tracking ID: [EXPOSUREAPP-5007](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5007)

### CODE REVIEW

**Comment by dsarkar** [comment]
Internal tracking ID: [EXPOSUREAPP-5007](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5007)

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml [hunk 1]
```diff
@@ -109,6 +109,8 @@
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/spacing_small"
                         android:text="@string/submission_contact_body_other"
+                        android:autoLink="phone"
+                        android:textColorLink="@color/colorTextTint"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:113]: Thanks for this. I think it does not work for All Android versions.
You fix worked on Pixel 4a - Android 11. 
I have tested it on S8 Android 9 and emulator Galaxy Nexus Android 6 and the fix is not working on these versions.
We will have to do programatically in this case.
 maybe an extension function to be used in other places too:
```kotlin
fun TextView.linkifyPhoneNumbers(){
    LinkifyCompat.addLinks(
        this,
        Patterns.PHONE,
        "tel:",
        Linkify.sPhoneNumberMatchFilter,
        Linkify.sPhoneNumberTransformFilter
    )
    movementMethod = LinkMovementMethod.getInstance()
    setLinkTextColor(context.getColorCompat(R.color.colorTextTint))
}
```

Comment (MikeMcC399) [reply Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:113]: @mtwalli 
Thanks for your feedback and it confirms what I found in https://stackoverflow.com/questions/40788608/androidautolink-for-phone-numbers-doesnt-always-work which shows how the effect of autoLink depends on the device's locale.

In a Nexus 6 emulator, Android 6 (API 23) with the device language set to English (United States) the telephone number 116 117 in the App Information > Technical Hotline view (which this PR does not change) is **not** linked. I did also a spot check on a Samsung Galaxy A5 (2017), Android 8 when the locale is English (United States) the telephone number 116 117 is not linked. It does appear linked when the language is changed to English (United Kingdom). 

Using a Pixel 3a emulator, Android 11 I got a different result: the telephone number 116 117 was never shown linked, independently of the locale. I saw in a short test program that Android 11 only ever linked 7-digit and 10-digit telephone numbers and international phone numbers like +49 xx in a TextView.

So the conclusion is that autoLink for phone numbers is not reliable.

The changes I proposed in this PR simply align the Technical Hotline view with the Request TAN view.

If the phone number 116 117 in the Technical Hotline view is shown linked in a certain environment, then the phone number 116 117 in the Request TAN view is also shown linked. If the link does not work in the Technical Hotline view, it will consistently not be working in the Request TAN view either.

The issue is much more complex than I imagined and a better implementation may need to be deferred. LinkifyCompat looks like it would provide a general solution. For the single phone number "Medical Emergency Service Hotline" 116 117 it may be simpler to make it a special case and implement a calling button or similar.

In summary, the changes proposed in this PR improve the situation for the Request TAN view in some environments and don't make it any worse. The changes are however not sufficient to make the phone number 116 117 clickable in all environments.

Comment (mtwalli) [reply Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:113]: @MikeMcC399 Thank you for the detailed information.
Based on our discussion on https://github.com/corona-warn-app/cwa-app-android/pull/2286
@PhilippNowak96  kindly implemented the proposed solution there, if you confirm that the changes in https://github.com/corona-warn-app/cwa-app-android/pull/2286 fixes it on the devices you mentioned above, can we close this PR in favour of https://github.com/corona-warn-app/cwa-app-android/pull/2286?

**Comment by MikeMcC399** [comment]
@mtwalli 

As noted in https://github.com/corona-warn-app/cwa-app-android/pull/2286#issuecomment-774646592 PR "Fix autoLink being too aggressive on some devices (closes #2275) (EXPOSUREAPP-4992)" #2286 mitigates the issue "Medical Emergency Service Hotline 116 117 not linked as phone in Request TAN" #2291 and makes this PR obsolete. 

I'll leave this issue open for the moment and plan to close it.

I'm sorry for the delay in replying. I rely a lot on e-mail notification and your last comment in the review two days ago didn't trigger a notification. (I guess it would have had to be a new review request for that to happen.)

I hope that #2286 can be merged in time for the 1.12.0 release so that the linking issue are all fixed.

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml (hunk 1 above)
Comment (MikeMcC399) [reply Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:113]: I agree that this PR can be closed in favour of #2286.

**Comment by MikeMcC399** [comment]
Closed in favour of #2286.
